tests/integration_test.rs

```diff
@@ -79,39 +79,21 @@ async fn integration_test_hardcoded_pipeline() {
             .unwrap()
             .into_inner();
 
-        let mut step_recv_count = 0;
-        let mut spike_recv_count = 0;
-        let mut buddy_recv_count = 0;
-        let mut sct_recv_count = 0;
-        while let Some(recv) = stream.next().await {
-            let inner = recv.unwrap();
-            match inner.test.as_ref() {
-                "spike_check" => {
-                    spike_recv_count += 1;
-                }
-                "step_check" => {
-                    step_recv_count += 1;
-                }
-                "buddy_check" => {
-                    buddy_recv_count += 1;
-                }
-                "sct" => {
-                    sct_recv_count += 1;
-                }
-                _ => {
-                    panic!("unrecognised test name returned")
-                }
-            }
-            let flags: Vec<i32> = inner.results.iter().map(|res| res.flag).collect();
-            assert!(
-                flags == vec![Flag::Pass as i32; DATA_LEN_SPATIAL]
-                    || flags == vec![Flag::Isolated as i32; DATA_LEN_SPATIAL]
-            );
-        }
-        assert_eq!(spike_recv_count, 1);
-        assert_eq!(step_recv_count, 1);
-        assert_eq!(buddy_recv_count, 1);
-        assert_eq!(sct_recv_count, 1);
+        assert_eq!(response.results.len(), 4);
+        assert!(response.results.iter().any(|c| c.check == "step_check"));
+        assert!(response.results.iter().any(|c| c.check == "spike_check"));
+        assert!(response.results.iter().any(|c| c.check == "buddy_check"));
+        assert!(response.results.iter().any(|c| c.check == "sct"));
```

Comment:
Should we check that the order is correct?

Reply:
While I'm fairly certain the ordering will be consistent, I'm not sure it matters? Any ordering will be indistinguishable once it hits the DB right?

In any case, this integration test could probably use a glow-up soon with how much has been fundamentally changing since I first wrote it

Reply:
Ok, we can take it later then